homeassistant/components/tplink/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""Component to embed TP-Link smart home devices."""
+import logging
+
+from homeassistant.const import CONF_HOST
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_flow
+import homeassistant.helpers.config_validation as cv
+import voluptuous as vol
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'tplink'
+
+TPLINK_HOST_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string
+})
+
+CONF_LIGHT = 'light'
+CONF_SWITCH = 'switch'
+CONF_DISCOVERY = 'discovery'
+
+ATTR_CONFIG = 'config'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional('light', default=[]): vol.All(cv.ensure_list,
+                                                   [TPLINK_HOST_SCHEMA]),
+        vol.Optional('switch', default=[]): vol.All(cv.ensure_list,
+                                                    [TPLINK_HOST_SCHEMA]),
+        vol.Optional('discovery', default=True): cv.boolean,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+REQUIREMENTS = ['pyHS100==0.3.3']
+
+
+async def _async_has_devices(hass):
+    """Return if there are devices that can be discovered."""
+    from pyHS100 import Discover
+
+    def discover():
+        devs = Discover.discover()
+        return devs
+    return await hass.async_add_executor_job(discover)
+
+
+async def async_setup(hass, config):
+    """Set up the TP-Link component."""
+    conf = config.get(DOMAIN)
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][ATTR_CONFIG] = conf
+
+    hass.async_create_task(hass.config_entries.flow.async_init(
```

Comment:
We only want to do this if `conf` is not `None`.

Reply:
Ah, it used to do that, but I thought there will always be a config available (when defined in the file). I'll change it back.

Reply:
Changed.